### PR TITLE
Added Makefile tutorial

### PR DESCRIPTION
**Pull request by Existential-nonce**
## What does this PR do?
Add resource(s)

### Description
Added the makefile cookbook tutorial from https://makefiletutorial.com/

### Why is this valuable (or not)?
Makefile should be a fundamental tool for C/C++ developers.

### How do we know it's really free?
It’s open source and doesn’t require any credentials, emails or anything.

### For book lists, is it a book? For course lists, is it a course? etc.
It’s a github repository.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Followup

- Check the output of Travis-CI for linter errors!

### CODE REVIEW

**Comment by Existential-nonce** [comment]
Not sure why it’s not building properly, I’m pretty sure I’ve checked the syntax of the PR and it should be good. I’m assuming this has to do with the link.

**Comment by eshellman** [comment]
Be sure to check the output of Travis-CI for linter errors. Alphabetical order and spacing are the most common problems. (might be trailing slash)

File: books/free-programming-books.md [hunk 1]
```diff
@@ -2121,6 +2121,7 @@ Kerridge (PDF) (email address *requested*, not required)
 
 ### Make
 
+* [Makefile tutorial](https://makefiletutorial.com/)
```
Comment (davorpa) [inline books/free-programming-books.md:2124]: - Give creator credits
- Fix lint errors (https://app.travis-ci.com/github/EbookFoundation/free-programming-books/builds/238391859#L217-L222)
  ```log
  $ fpb-lint ./books/
  /home/travis/build/EbookFoundation/free-programming-books/books/free-programming-books.md
    2124:3-2124:53  warning  Remove trailing slash (https://makefiletutorial.com)  no-url-trailing-slash  remark-lint
  ⚠ 1 warning
  The command "fpb-lint ./books/" exited with 1.
  ```

```suggestion
* [Makefile tutorial](https://makefiletutorial.com) - Chase Lambert
```

**Review by davorpa — CHANGES_REQUESTED** [review]
Nice tutorial/cookbook @theicfire :rocket:

**Comment by Existential-nonce** [comment]
I forgot to revert my commit where I credited the author of the repository. My bad. I was testing if it was going to pass the build if I removed anything other than the title and the hyperlink.

**Review by davorpa — APPROVED** [review]

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Comment by Existential-nonce** [comment]
No problem, I’ll most likely contribute more resources to the repository in the near future.

**Comment by theicfire** [comment]
Oh neat, cheers!

**Pull request merged**